### PR TITLE
TN-662 refactor inactivity timeout

### PR DESCRIPTION
**Pull request by pbugni**
See details in document https://docs.google.com/document/d/1LAXMyS0M4ttlArIhPQc6fVuB3oXN_XAI0HqCsSa2VdE/edit#

@achen2401 I think (given the complexity of http cookies) we might want to set `data-cookie-timeout` to the template variable `{{ expires_in }}` in `portal_wrapper.html` ??

### CODE REVIEW

**Comment by achen2401** [comment]
@pbugni paul, I like the idea of using the {{expires_in}} template variable after looking over the code.  It Is a cleaner approach.  I will see if i can refactor the frontend code tomorrow morning when my thinking won’t be as fuzzy.

File: portal/config/config.py [hunk 1]
```diff
@@ -93,8 +93,10 @@ class BaseConfig(object):
     LOG_LEVEL = 'DEBUG'
 
     OAUTH2_PROVIDER_TOKEN_EXPIRES_IN = 4 * 60 * 60  # units: seconds
-    SS_TIMEOUT = 60 * 60  # seconds for session cookie, reset on ping
-    PERMANENT_SESSION_LIFETIME = SS_TIMEOUT
+    DEFAULT_INACTIVITY_TIMEOUT = 30 * 60  # default inactivity timeout
+    PERMANENT_SESSION_LIFETIME = 60 * 60  # defines life of redis session
```
Comment (ivan-c) [inline portal/config/config.py]: This also sets the lifetime of the cookie FYI
http://flask.pocoo.org/docs/1.0/config/#PERMANENT_SESSION_LIFETIME

Comment (pbugni) [reply portal/config/config.py]: Yes, but that's fine.  You'll note the cookie used to retain the /settings value is given a different lifetime.

File: portal/static/js/sessionMonitor.js [hunk 2]
```diff
@@ -37,107 +20,113 @@ var SessionMonitorObj = function() {
...
                     onwarning: function() {
-                      var n = Math.round(l.timeBeforeWarning / 60 / 1e3),
-                          o = $('<div id="jqsm-warning">Your session will expire in ' + n + ' minutes. <button id="jqsm-stay-logged-in">Stay Logged In</button><button id="jqsm-log-out">Log Out</button></div>');
-                      $("body").children("div#jqsm-warning").length || $("body").prepend(o), $("div#jqsm-warning").show(), $("button#stay-logged-in").on("click", function(n) {
-                          n && n.stopPropagation(), l.extendsess(n);
-                      }).on("click", function() {
-                          o.hide();
-                      }), $("button#jqsm-log-out").on("click", l.logout);
+                        var n = Math.round(l.timeBeforeWarning / 60 / 1e3),
+                            o = $("<div id='jqsm-warning'>Your session will expire in " + n + " minutes. <button id='jqsm-stay-logged-in'>Stay Logged In</button><button id='jqsm-log-out'>Log Out</button></div>");
```
Comment (ivan-c) [inline portal/static/js/sessionMonitor.js]: Should this be translated?

Comment (achen2401) [reply portal/static/js/sessionMonitor.js]: @ivan-c ivan, this is default html if custom one is not provided - which we do provide and the custom html is tagged for translation.

**Pull request merged**